test/getUsers.spec.js

```diff
@@ -0,0 +1,98 @@
+const supertest = require('supertest');
+const { expect } = require('chai');
+const app = require('../app');
+const request = supertest(app);
+const { createUsers } = require('./helpers');
+const { host, port } = require('../config').common.api;
+const { AUTHENTICATION_ERROR, NOT_FOUND_ERROR } = require('../app/errors');
+
+describe('GET /users', () => {
+  it('should success when user is logged in', () =>
+    createUsers(25)
+      .then(() =>
+        request.post('/users/sessions').send({ email: 'mail1@wolox.com.ar', password: '1234587ocho' })
+      )
+      .then(response => {
+        const token = response.headers.authorization;
+        const nextRequest = request.get('/users').set({ authorization: token });
+        return Promise.all([nextRequest, token]);
+      })
+      .then(([response, token]) => {
+        const { count, result, prev, next } = response.body;
+        expect(response.statusCode).to.equal(200);
+        expect(count).to.equal(25);
+        expect(result.length).to.equal(10);
+        expect(prev).to.equal(null);
+        expect(next).to.equal(`${host}:${port}/users?page=2`);
+        const nextRequest = request.get('/users?page=2').set({ authorization: token });
+        return Promise.all([nextRequest, token]);
+      })
+      .then(([response, token]) => {
+        const { count, result, prev, next } = response.body;
+        expect(response.statusCode).to.equal(200);
+        expect(count).to.equal(25);
+        expect(result.length).to.equal(10);
+        expect(prev).to.equal(`${host}:${port}/users?page=1`);
+        expect(next).to.equal(`${host}:${port}/users?page=3`);
+
+        return request.get('/users?page=3').set({ authorization: token });
+      })
+      .then(response => {
+        const { count, result, prev, next } = response.body;
+        expect(response.statusCode).to.equal(200);
+        expect(count).to.equal(25);
+        expect(result.length).to.equal(5);
+        expect(prev).to.equal(`${host}:${port}/users?page=2`);
+        expect(next).to.equal(null);
```

Comment:
You could create a function for make request (passing page) and another for check response for simplify a little bit this test